src/flow.c

```diff
@@ -1131,7 +1131,8 @@ uint8_t FlowGetDisruptionFlags(const Flow *f, uint8_t flags)
     TcpSession *ssn = f->protoctx;
     TcpStream *stream = flags & STREAM_TOSERVER ? &ssn->client : &ssn->server;
 
-    if (stream->flags & STREAMTCP_STREAM_FLAG_DEPTH_REACHED) {
+    if ((stream->flags & STREAMTCP_STREAM_FLAG_DEPTH_REACHED) ||
+            (ssn->flags & STREAMTCP_FLAG_APP_LAYER_DISABLED)) {
```

Comment:
I think this makes sense, but I feel like the `STREAM_DEPTH` flags name is now getting confusing. Since we're using it as a "stream is interupted/disrupted/truncated" flag, I think we should also rename it. How about `STREAM_DISRUPTED`?

Reply:
ok doing